ambry-api/src/main/java/com.github.ambry/account/AclService.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.account;
+
+import com.github.ambry.rest.SecurityService;
+
+
+/**
+ * A service that handles authorization of requests. This will be provided to the {@link SecurityService} for making
+ * access decisions that might require the help of an external service. The call to
+ * {@link #hasAccess(Object, Resource, Operation)} should not require expensive network calls, since it will potentially
+ * be used on the critical path. Instead, the {@link AclService} should maintain a cache of ACLs for the resources it
+ * manages permissions on.
+ * @param <P> the type for the principal. This is generic to allow for different requester authentication schemes.
+ */
+public interface AclService<P> {
+
+  /**
+   * Makes a resource access decision.
+   * @param principal the requester principal (identity).
+   * @param targetResource the {@link Resource} to check for access to.
+   * @param operation the {@link Operation} to perform on the resource.
+   * @return an {@link AccessDecision} for the principal performing the specified operation on the target resource.
+   */
+  AccessDecision hasAccess(P principal, Resource resource, Operation operation);
```

Comment:
I was thinking that we might want to add additional status codes in the future to indicate cases where the service might be unavailable or the ACL might not be found. The alternative is to leave this up to the AclService implementer to set this behavior (and log appropriately in cases like those listed above). I think I will change this to a boolean